kotlin/samples/containers/app-poetry/src/main/java/com/squareup/sample/poetryapp/PoemsBrowserWorkflow.kt

```diff
@@ -0,0 +1,47 @@
+package com.squareup.sample.poetryapp
+
+import com.squareup.sample.container.masterdetail.MasterDetailScreen
+import com.squareup.sample.poetry.PoemWorkflow
+import com.squareup.sample.poetry.model.Poem
+import com.squareup.workflow.RenderContext
+import com.squareup.workflow.Snapshot
+import com.squareup.workflow.StatefulWorkflow
+import com.squareup.workflow.WorkflowAction
+import com.squareup.workflow.parse
+import com.squareup.workflow.ui.BackStackScreen
+import com.squareup.workflow.workflowAction
+
+object PoemsBrowserWorkflow : StatefulWorkflow<List<Poem>, Int, Nothing, MasterDetailScreen>() {
+  override fun initialState(
+    props: List<Poem>,
+    snapshot: Snapshot?
+  ): Int {
+    return snapshot?.bytes?.parse { source -> source.readInt() }
+        ?: -1
+  }
+
+  override fun render(
+    props: List<Poem>,
+    state: Int,
+    context: RenderContext<Int, Nothing>
+  ): MasterDetailScreen {
+    val poems = context.renderChild(PoemListWorkflow, props) { selected -> choosePoem(selected) }
+        .copy(selection = state)
+        .let { BackStackScreen<Any>(it) }
+
+    return if (state == -1) {
+      MasterDetailScreen(poems)
+    } else {
+      MasterDetailScreen(poems) + context.renderChild(PoemWorkflow, props[state]) { choosePoem(-1) }
```

Comment:
This `+` operator almost seems too magic to me. There's too much going on, since this type is a pair of lists. Does this operator concatenate to a new pair? Or is it a distributed concatenation of all the component lists? We can add a bunch of comments but that's already a smell. I'd rather have a method like `appendMasterDetail` or something, to make it clear that it's operating on both and what it's doing.

Reply:
Alternatively we can implement it as some infix function, though at this point you might as well just have a regular function.

Reply:
Actually I would also probably pull this render call out into a val so the screen composition is more clear.

Reply:
"combine" isn't really any clearer to me. But I don't have any better suggestions, might as well just use the `+` operator. Extracting the render call makes me a lot more comfortable in any case.

Reply:
Yay! Restoring plus.

Reply:
```
    val poems: MasterDetailScreen =
      context.renderChild(PoemListWorkflow, props) { selected -> choosePoem(selected) }
          .copy(selection = state)
          .let { MasterDetailScreen(BackStackScreen(it)) }

    return if (state == -1) {
      poems
    } else {
      val poem: MasterDetailScreen =
        context.renderChild(PoemWorkflow, props[state]) { clearSelection }
      poems + poem
    }
```